### PR TITLE
Add ORC writing test cases for not implicitly lowercase columns [databricks]

### PR DESCRIPTION
**Pull request by res-life**
closes https://github.com/NVIDIA/spark-rapids/issues/8793

Add a test case for ORC writing.

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Review by revans2 — DISMISSED** [review]

File: integration_tests/src/main/python/orc_write_test.py [hunk 1]
```diff
@@ -15,11 +15,11 @@
 import pytest
 
 from asserts import assert_gpu_and_cpu_writes_are_equal_collect, assert_gpu_fallback_write
-from spark_session import is_before_spark_320, is_spark_cdh, is_spark_321cdh
+from spark_session import is_before_spark_320, is_spark_cdh, is_spark_321cdh, with_gpu_session, with_cpu_session
```
Comment (mythrocks) [inline integration_tests/src/main/python/orc_write_test.py:18]: Nitpick: Should the imports be alphabetically sorted in the same line?

File: integration_tests/src/main/python/orc_write_test.py [hunk 2]
```diff
@@ -15,11 +15,11 @@
...
 from datetime import date, datetime, timezone
 from data_gen import *
 from marks import *
-from pyspark.sql.functions import lit
+from pyspark.sql.functions import lit, col
```
Comment (mythrocks) [inline integration_tests/src/main/python/orc_write_test.py:22]:
```suggestion
from pyspark.sql.functions import col, lit
```

File: integration_tests/src/main/python/orc_write_test.py [hunk 3]
```diff
@@ -281,3 +281,27 @@ def test_orc_write_column_name_with_dots(spark_tmp_path):
...
+        # column is uppercase
+        lambda spark, path: spark.range(0, 1000).select(col("id").alias("Acol")).write.orc(path),
+        lambda spark, path: spark.read.orc(path),
+        data_path)
+    try:
+        # reading lowercase causes exception
+        with_cpu_session(lambda spark: spark.read.orc(data_path + "/CPU").schema["acol"])
+        assert False
+    except KeyError as e:
+        assert ("No StructField named acol" in str(e))
```
Comment (mythrocks) [inline integration_tests/src/main/python/orc_write_test.py:301]: We can keep this uniform with the GPU run below.
```suggestion
        assert "No StructField named acol" in str(e)
```

**Review by mythrocks — DISMISSED** [review]
Very minor nitpicks. Looks good to me!

**Comment by res-life** [comment]
build

File: integration_tests/src/main/python/orc_write_test.py (hunk 1 above)
Comment (res-life) [reply integration_tests/src/main/python/orc_write_test.py]: Done

File: integration_tests/src/main/python/orc_write_test.py (hunk 2 above)
Comment (res-life) [reply integration_tests/src/main/python/orc_write_test.py]: Done

File: integration_tests/src/main/python/orc_write_test.py (hunk 3 above)
Comment (res-life) [reply integration_tests/src/main/python/orc_write_test.py]: Done

**Review by revans2 — APPROVED** [review]

**Pull request merged**